src/relax/transform/gradient.cc

```diff
@@ -0,0 +1,407 @@
+      // b_adjoint_expr += a_adjoint_var[0], c_adjoint_expr += a_adjoint_var[1]
+      UpdateExprMap(GetRef<Tuple>(node), adjoint_expr_map_[binding->var]);
+    } else if (const auto* node = binding->value.as<TupleGetItemNode>()) {
+      // case 2: tuple get item
+      // b = a[0]
+      // a_adjoint_expr[0] (in fields) += b_adjoint_var
+      // a = ((x, y), (z,))
+      // b = a[0]
+      UpdateExprMap(GetRef<TupleGetItem>(node), adjoint_expr_map_[binding->var]);
+    } else if (const auto* node = binding->value.as<VarNode>()) {
+      // case 3: assign
+      // a = b
+      // b_adjoint_expr += a_adjoint_var
+      UpdateExprMap(GetRef<Var>(node), adjoint_expr_map_[binding->var]);
+    } else if (const auto* node = binding->value.as<CallNode>()) {
+      // case 4: call
+      const Op& call_op = GetRef<Op>(node->op.as<OpNode>());
+      const Array<Expr>& partials = gradient_op_map[call_op](GetRef<Call>(node), adjoint_var);
+      ICHECK(partials.size() == node->args.size()) << "partials number != inputs number";
+      for (size_t i = 0; i < partials.size(); ++i) {
+        const VarNode* arg = node->args[i].as<VarNode>();
+        ICHECK(arg != nullptr);
+        UpdateExprMap(GetRef<Var>(arg), partials[i]);
+      }
+    } else {
+      LOG(FATAL) << "AD does not support this type of binding value now: " << binding->value;
+    }
+  }
+
+ private:
+  Expr ReplaceExprByVar(Expr expr) {
+    if (adjoint_expr_to_var_.count(expr)) {
+      return adjoint_expr_to_var_[expr];
+    }
+    return expr;
+  }
+
+  void CreateAdjointVar(const Var& v, bool is_dataflow_var) {
+    // the adjoint var has been created
+    if (adjoint_var_map_.count(v)) return;
+    if (is_dataflow_var) {
+      Var adjoint = DataflowVar(v->name_hint() + "_adjoint", v->shape(), v->checked_type());
+      adjoint_var_map_.Set(v, adjoint);
+    } else {
+      Var adjoint = Var(v->name_hint() + "_adjoint", v->shape(), v->checked_type());
+      adjoint_var_map_.Set(v, adjoint);
+    }
+  }
+
+  // base could only be one of these types:
+  // Var, Tuple, TupleGetItem
+  void UpdateExprMap(const Expr& base, const Expr& increment) {
+    if (const auto* node = base.as<VarNode>()) {
+      const Var& v = GetRef<Var>(node);
+      if (adjoint_expr_map_.count(v) == 0) {
+        adjoint_expr_map_.Set(v, ReplaceExprByVar(increment));
+      } else {
+        const Expr& updated = DoAdd(adjoint_expr_map_[v], increment);
+        adjoint_expr_map_.Set(v, updated);
+      }
+    } else if (const auto* node = base.as<TupleNode>()) {
+      if (const auto* node1 = increment.as<TupleNode>()) {
+        for (size_t i = 0; i < node->fields.size(); ++i) {
+          UpdateExprMap(node->fields[i], node1->fields[i]);
+        }
+      } else {
+        LOG(FATAL) << "base and increment should be both tuple";
+      }
+    } else if (const auto* node = base.as<TupleGetItemNode>()) {
+      ICHECK(node->tuple->IsInstance<VarNode>()) << "Tuple of TupleGetItem must be binded to a Var";
```

Comment:
The past tense of “bind” is “bound” :-)
```suggestion
      ICHECK(node->tuple->IsInstance<VarNode>()) << "Tuple of TupleGetItem must be bound to a Var";
```